csharp/src/Google.Protobuf/WellKnownTypes/TimestampPartial.cs

```diff
@@ -222,6 +222,93 @@ internal static string ToJson(long seconds, int nanoseconds, bool diagnosticOnly
             }
         }
 
+        /// <summary>
+        /// Given another timestamp, returns 0 if the timestamps are equivalent, -1 if this timestamp precedes the other, and 1 otherwise
+        /// </summary>
+        /// <param name="other">Timestamp to compare</param>
+        /// <returns>an integer indicating whether this timestamp precedes or follows the other</returns>
+        public int CompareTo(Timestamp other)
```

Comment:
Does CompareTo needs to be public? I'd prefer to make it private if possible and only expose < == operations etc. to users

Reply:
Well, we are implementing ```IComparable```, so does it have to be public?  It does have to be exposed, yes.  We could make the implementation explicit, but it will always be public on the interface and therefore always exposed to users and so there's no way around exposing it.  Note that for reference, these are always public in the BCL, as far as I can tell.